.github/workflows/phpunit.yml

```diff
@@ -44,6 +45,9 @@ jobs:
           key: "php-${{ matrix.php-version }}-${{ matrix.dependencies }}"
           restore-keys: "php-${{ matrix.php-version }}-${{ matrix.dependencies }}"
 
+      - name: "Remove unneded dep with matrix incompatibilities"
```

Comment:
This step directly collides with using "locked" dependencies, since `composer remove` modifies `composer.lock`.

As an alternative, let's move `roave/backward-compatibility-check` to a `tools/` directory where `composer.json` and `composer.lock` are configured for it, and we exclude it in `.gitattributes`

Reply:
Done